src/kv/snapshotter.h

```diff
@@ -7,16 +7,18 @@
 #include "consensus/ledger_enclave_types.h"
 #include "ds/thread_messaging.h"
 #include "kv/kv_types.h"
+#include "kv/snapshot_serdes.h"
```

Comment:
I think it'd be better to leave the snapshotter out of `kv/`. The kv component only needs to know about the `AbstractSnapshotter` type, which is implemented by the `ccf::Snapshotter`.